packages/@glimmer/runtime/lib/helpers/invoke.ts

```diff
@@ -53,51 +55,27 @@ export function invokeHelper(
     );
   }
 
-  const owner = getOwner(context);
-  const internalManager = getInternalHelperManager(definition)!;
+  let helper = getInternalHelperManager(definition)!;
+  let owner = getOwner(context);
 
   // TODO: figure out why assert isn't using the TS assert thing
-  if (DEBUG && !internalManager) {
-    throw new Error(
-      `Expected a helper definition to be passed as the second parameter to invokeHelper, but no helper manager was found. The definition value that was passed was \`${debugToString!(
-        definition
-      )}\`. Did you use setHelperManager to associate a helper manager with this value?`
-    );
+  assert(
+    helper,
+    `Expected a helper definition to be passed as the second parameter to invokeHelper, but no helper manager was found. The definition value that was passed was \`${debugToString!(
+      definition
+    )}\`. Did you use setHelperManager to associate a helper manager with this value?`
+  );
+
+  if (typeof helper !== 'function') {
+    helper = helper.getHelper(definition);
   }
 
-  if (DEBUG && typeof internalManager === 'function') {
-    throw new Error(
-      'Found a helper manager, but it was an internal built-in helper manager. `invokeHelper` does not support internal helpers yet.'
-    );
-  }
-
-  const manager = (internalManager as InternalHelperManager<object>).getDelegateFor(owner);
   let args = new SimpleArgsProxy(context, computeArgs);
-  let bucket = manager.createHelper(definition, args);
-
-  let cache: Cache<unknown>;
+  let helperRef = helper(args as CapturedArguments, owner);
```

Comment:
Double check this, `args` is not a `CapturedArguments` (its a proxy to values, not a bunch of references).

Also need tests.